src/model/lang.ml

```diff
@@ -555,10 +555,24 @@ let umty_of_mty : ModuleType.expr -> ModuleType.U.expr option = function
 (** Query the top-comment of a signature. This is [s.doc] most of the time with
     an exception for signature starting with an inline includes. *)
 let extract_signature_doc (s : Signature.t) =
+  let rec uexpr_is_hidden = function
+    | ModuleType.U.Path p -> Path.is_hidden (p :> Path.t)
+    | Signature _ ->
+        true (* Hidden in some sense, we certainly want its top comment *)
+    | With (_, e) -> uexpr_is_hidden e
+    | TypeOf { t_desc = ModPath p | StructInclude p; _ } ->
+        Path.is_hidden (p :> Path.t)
+  in
   match (s.doc, s.items) with
   | [], Include { expansion; status = `Inline; _ } :: _ ->
       (* A signature that starts with an [@inline] include inherits the
          top-comment from the expansion. This comment is not rendered for
          [include] items. *)
       expansion.content.doc
+  | [], Include { expansion; decl = Alias p; _ } :: _
```

Comment:
I think this would be better written as a `should_inline_expansion` function to avoid duplicating the match on the doc, `Include` and the RHS of `->`.